lib/NormalModuleFactory.js

```diff
@@ -1020,6 +1020,10 @@ If changing the source code is not an option there is also a resolve options cal
 						const parsedResult = identToLoaderRequest(result);
 						const resolved = {
 							loader: parsedResult.loader,
+							type:
```

Comment:
I think you also want to pass `"module"` when `/\.mjs$/i.test(parsedResult.loader)`.

And when it ends with `.cjs` you don't want to pass `"module"` even if the `descriptionFileData` contains that type